### PR TITLE
Add missing is_bitfield field for global enum in extension_api_dump

### LINKED ISSUE #71397: GDExtension: Global enum are not marked as bitfield in extension_api.json
### Godot version

4.0beta12

### System information

all

### Issue description

In `extension_api.json` we have some function that uses parameter typed after a global enum and display it as a bitfield:
```js
{
	"name": "set_button_mask",
	"is_const": false,
	"is_vararg": false,
	"is_static": false,
	"is_virtual": false,
	"hash": 3950145251,
	"arguments": [
		{
			"name": "mask",
			"type": "bitfield::MouseButtonMask" // bitfield prefix here !
		}
	]
},
```

However the global enum's own definition doesn't mention it is a bitfield:
```js
{
	"name": "MouseButtonMask",
	// No "bitfield" field here !
	"values": [
		{
			"name": "MOUSE_BUTTON_MASK_LEFT",
			"value": 1
		},
		{
			"name": "MOUSE_BUTTON_MASK_RIGHT",
			"value": 2
		},
		{
			"name": "MOUSE_BUTTON_MASK_MIDDLE",
			"value": 4
		},
		{
			"name": "MOUSE_BUTTON_MASK_MB_XBUTTON1",
			"value": 128
		},
		{
			"name": "MOUSE_BUTTON_MASK_MB_XBUTTON2",
			"value": 256
		}
	]
},
```

On the other hand, the non-global enums (i.e. the ones defined as part of a class) have this `is_bitfield` field.
So I guess we should add it to global enum too.

### Steps to reproduce

n/a

### Minimal reproduction project

n/a

### PR DESCRIPTION
**Pull request by touilleMan**
fix #71397

### CODE REVIEW

**Comment by raulsntos** [comment]
- Follow up to https://github.com/godotengine/godot/pull/71045

File: core/extension/extension_api_dump.cpp [hunk 1]
```diff
@@ -458,6 +458,7 @@ Dictionary GDExtensionAPIDump::generate_extension_api() {
 		// Global enums and constants.
 		Array constants;
 		HashMap<String, List<Pair<String, int64_t>>> enum_list;
+		HashMap<String, bool> enum_is_bitfield;
```
Comment (raulsntos) [inline core/extension/extension_api_dump.cpp:461]: Alternatively, we could create a struct like:

```c++
struct EnumInterface {
	String name;
	bool is_bitfield;
	List<Pair<String, int64_t>> values;
};
```

Then we'd only need one map:

```c++
HashMap<String, EnumInterface> enum_list;
```

Comment (touilleMan) [reply core/extension/extension_api_dump.cpp:461]: I'd rather stick to the current solution, it is in theory less performant (two time more hashmap insert/lookups) but this code is not performance critical so it's no big deal. On the other hand avoiding a new struct declaration make the code a bit simpler (no need to jump between struct declaration and code using it).

But that's not a very strong take, if you really prefere it I will make the change @raulsntos  😃

Comment (raulsntos) [reply core/extension/extension_api_dump.cpp:461]: I'm fine either way so whatever is preferred by the team.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**